src/Microsoft.Health.Fhir.SqlServer/Features/Operations/Import/DataGenerator/DateTimeSearchParamsTableBulkCopyDataGenerator.cs

```diff
@@ -72,5 +75,61 @@ internal override void FillSchema(DataTable table)
             table.Columns.Add(new DataColumn(VLatest.DateTimeSearchParam.IsMin.Metadata.Name, VLatest.DateTimeSearchParam.IsMin.Metadata.SqlDbType.GetGeneralType()));
             table.Columns.Add(new DataColumn(VLatest.DateTimeSearchParam.IsMax.Metadata.Name, VLatest.DateTimeSearchParam.IsMax.Metadata.SqlDbType.GetGeneralType()));
         }
+
+        internal static IEnumerable<BulkDateTimeSearchParamTableTypeV2Row> Distinct(IEnumerable<BulkDateTimeSearchParamTableTypeV2Row> input)
+        {
+            return input.Distinct(Comparer);
+        }
+
+        internal class BulkDateTimeSearchParamTableTypeV2RowComparer : IEqualityComparer<BulkDateTimeSearchParamTableTypeV2Row>
+        {
+            public bool Equals(BulkDateTimeSearchParamTableTypeV2Row x, BulkDateTimeSearchParamTableTypeV2Row y)
+            {
+                if (x.SearchParamId != y.SearchParamId)
+                {
+                    return false;
+                }
+
+                if (!x.StartDateTime.Equals(y.StartDateTime))
```

Comment:
Instead of using the object's Equal method, should we use the static DateTimeOffset.Equals method to avoid potential null pointer exception?

Reply:
Actually the StartDateTime or EtartDateTime will not be null according to the .sql definition, but from c# code perspective, I think your advice is better, so I turned to use DateTimeOffset.Equals instead.